### PR TITLE
[XLA:CPU][oneDNN] Minor fix in CPU bfloat16 support

### PR DESCRIPTION
**Pull request by yimeisun123**
Align the usage of oneDNN bf16 cpu float normalization with the same guarding condition as for oneDNN rewrites in compilation step, i.e., only when oneDNN is enabled and in non-AOT mode.

### CODE REVIEW

**Review by tdanyluk — APPROVED** [review]
I'm not a CPU expert, but this looks fine. Thank you for the contribution.

File: xla/service/cpu/cpu_compiler.cc [hunk 1]
```diff
@@ -711,12 +711,17 @@ Status CpuCompiler::RunHloPassesThroughLayoutAssn(
   // Convert BF16 and F8 operations to F32 and F16 respectively so that the CPU
   // backend can support BF16/F8 operations without directly implementing a
   // BF16/F8 lowering for most ops.
+  FloatSupport bf16_support(BF16);
+  CpuFloatSupport onednn_bf16_support(BF16);
```
Comment (penpornk) [inline xla/service/cpu/cpu_compiler.cc:715]: `CpuFloatSupport` is only defined when `INTEL_MKL` and `ENABLE_ONEDNN_V3` are defined. It must always be under the `#if defined(INTEL_MKL) && defined(ENABLE_ONEDNN_V3)` guard.

See an example error message in the [ARM CI](https://source.cloud.google.com/results/invocations/55cc0324-7842-4614-950b-2cddf4073439/log) failure:
```
xla/service/cpu/cpu_compiler.cc:715:3: error: unknown type name 'CpuFloatSupport'; did you mean 'FloatSupport'?
  715 |   CpuFloatSupport onednn_bf16_support(BF16);
      |   ^~~~~~~~~~~~~~~
      |   FloatSupport
./xla/service/float_support.h:28:7: note: 'FloatSupport' declared here
   28 | class FloatSupport {
      |       ^
1 warning and 1 error generated.
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

**Comment by penpornk** [comment]
For future PRs, please also add a description on what the PR does and what problem it addresses, e.g., for this PR, we could say that there was a mismatch in adding expanded bf16 dtype support for oneDNN rewrite: oneDNN custom calls aren't used in AOT mode so the expanded CpuFloatSupport (which is specific to oneDNN custom calls) shouldn't be used in AOT mode too.

File: xla/service/cpu/cpu_compiler.cc (hunk 1 above)
Comment (yimeisun123) [reply xla/service/cpu/cpu_compiler.cc]: Thanks for this info. Updated with moving the onednn_bf16_support under the guard.

**Comment by yimeisun123** [comment]
Updated description field too. Please let me know if additional things need to be updated. Thanks for your review and suggestions.

**Review by penpornk — APPROVED** [review]
Thank you for the fix!

**Pull request closed without merging**